CMakeLists.txt

```diff
@@ -0,0 +1,119 @@
+cmake_minimum_required(VERSION 3.1)
+project(libsamplerate VERSION 0.1.9 LANGUAGES C)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)
+
+include(TestBigEndian)
+include(CheckFunctionExists)
+include(CheckIncludeFiles)
+include(CheckSymbolExists)
+include(CheckTypeSize)
+include(ClipMode)
+
+set(SAMPLERATE_SRC
+	${PROJECT_SOURCE_DIR}/src/samplerate.c
+	${PROJECT_SOURCE_DIR}/src/src_linear.c
+	${PROJECT_SOURCE_DIR}/src/src_sinc.c
+	${PROJECT_SOURCE_DIR}/src/src_zoh.c)
+
+if(WIN32)
+	set(OS_IS_WIN32 TRUE)
+	set(SAMPLERATE_SRC
+		${SAMPLERATE_SRC}
+		${PROJECT_SOURCE_DIR}/Win32/libsamplerate-0.def)
+	include_directories(Win32)
+endif()
+
+test_big_endian(CPU_IS_BIG_ENDIAN)
+if(CPU_IS_BIG_ENDIAN)
+	set(CPU_IS_LITTLE_ENDIAN 0)
+else()
+	set(CPU_IS_LITTLE_ENDIAN 1)
+endif()
+
+check_function_exists(pow RESULT)
+if(NOT RESULT)
+  list(APPEND CMAKE_REQUIRED_LIBRARIES m)
+endif()
+check_function_exists(alarm HAVE_ALARM)
+check_function_exists(lrint HAVE_LRINT)
+check_function_exists(lrintf HAVE_LRINTF)
+check_function_exists(signal HAVE_SIGNAL)
+
+check_include_files(stdint.h HAVE_STDINT)
+check_include_files(sys/times.h HAVE_SYS_TIMES_H)
+
+check_symbol_exists(SIGALRM signal.h HAVE_SIGALRM)
+
+check_type_size(int SIZEOF_INT)
+check_type_size(long SIZEOF_LONG)
+
+# This will set CPU_CLIPS_NEGATIVE and CPU_CLIPS_POSITIVE
+clip_mode()
+
+find_package(ALSA)
+set(HAVE_ALSA ${ALSA_FOUND})
+
+find_package(Sndfile)
+set(HAVE_SNDFILE ${SNDFILE_FOUND})
```

Comment:
@Bobo1239 cmake does not seem to provide `FindSndfile.cmake` or `FindFFTW.cmake` on some systems which means if they're in a non-standard location, the linking and including won't work as it would in a vanilla POSIX environment.

Would you consider adding an additional `pkg-config` dependency on this?  Then you could do the following:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 394de75..1b85d17 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ include(CheckIncludeFiles)
 include(CheckSymbolExists)
 include(CheckTypeSize)
 include(ClipMode)
+include(FindPkgConfig)

 set(SAMPLERATE_SRC
        ${PROJECT_SOURCE_DIR}/src/samplerate.c
@@ -53,12 +54,24 @@ clip_mode()

 find_package(ALSA)
 set(HAVE_ALSA ${ALSA_FOUND})
+if(ALSA_FOUND)
+       include_directories("${ALSA_INCLUDE_DIRS}")
+       link_directories("${ALSA_LIBRARY_DIRS}")
+endif()

-find_package(Sndfile)
+pkg_check_modules(Sndfile sndfile>=1.0.0)
 set(HAVE_SNDFILE ${SNDFILE_FOUND})
+if(SNDFILE_FOUND)
+       include_directories("${SNDFILE_INCLUDE_DIRS}")
+       link_directories("${SNDFILE_LIBRARY_DIRS}")
+endif()

-find_package(FFTW)
+pkg_check_modules(FFTW3 fftw3>=3.0.0)
 set(HAVE_FFTW3 ${FFTW_FOUND})
+if(FFTW3_FOUND)
+       include_directories("${FFTW3_INCLUDE_DIRS}")
+       link_directories("${FFTW3_LIBRARY_DIRS}")
+endif()

 configure_file(${PROJECT_SOURCE_DIR}/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
```

Reply:
`FindSndfile.cmake` and `FindFFTW.cmake` are the `cmake` directory which should be loaded by `set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)` at the top. (`ClipMode.cmake` is also affected by this.)

Regarding pkg-config: What do @erikd and @jcelerier think? Should I switch from the two cmake modules to pkg-config? I really don't know what is better supported across the board.

Reply:
>  I really don't know what is better supported across the board.

To give a bit more background information on this:

Ubuntu 14.04, default for Travis-CI:

```bash
locate -i FindALSA.cmake
(found) /usr/share/cmake-3.5/Modules/FindALSA.cmake
```

```bash
locate -i FindFFTW
(not found)
```

```
locate -i FindSndfile
(not found)
```

What this means is if a project requires any of the variables expected to be available for includes or libraries, they won't be available.  Msys2 re-uses the POSIX-style paths so it hides this bug as the default `include` and `lib` paths work out of the box.

Reply:
@Bobo1239 On second thought, perhaps this is a problem with my environment.  [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_PREFIX_PATH.html) clearly states that it searches the same paths as `pkg-config`.  Here's an excerpt from the website:

> *[CMAKE_PREFIX_PATH] contains the “base” directories, the `FIND_XXX()` commands append appropriate subdirectories to the base directories. So `FIND_PROGRAM()` adds `/bin` to each of the directories in the path, `FIND_LIBRARY()` appends `/lib` to each of the directories, and `FIND_PATH()` and `FIND_FILE()` append `/include` .*

I will investigate this some more on my end as to why the paths are found, but the includes and libs are not.

Reply:
I think you may have misunderstood me. FindFFTW and FindSndFile are distributed with the source (in the `cmake` directory) in the repo and aren't required from the environment.

Reply:
@Bobo1239 you are correct, I missed that they were distributed with the project. I think I found the culprit: https://github.com/erikd/libsamplerate/pull/24/files#r157309049.  Your original technique should be just fine but I recommend making the variables consistently named.  Also, `include_directories(...)` and `link_directories(...)` will still be required for some environments.

Reply:
I looked at some more `FindXXX.cmake` modules and they all appear to follow slightly different conventions.  Here's what works out of the box using this PR without adding `pkg-config`:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 394de75..a2aad7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,11 +53,23 @@ clip_mode()

 find_package(ALSA)
 set(HAVE_ALSA ${ALSA_FOUND})
+if(ALSA_FOUND)
+  include_directories("${ALSA_INCLUDE_DIR}")
+  link_directories("${ALSA_LIBRARY}/../")
+endif()

 find_package(Sndfile)
 set(HAVE_SNDFILE ${SNDFILE_FOUND})
+if(SNDFILE_FOUND)
+  include_directories("${SNDFILE_INCLUDE_DIR}")
+  link_directories("${SNDFILE_LIBRARY}/../")
+endif()

 find_package(FFTW)
+if(FFTW_FOUND)
+  include_directories("${FFTW_INCLUDES}")
+  link_directories("${FFTW_LIBRARIES}/../")
+endif()
 set(HAVE_FFTW3 ${FFTW_FOUND})

 configure_file(${PROJECT_SOURCE_DIR}/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
```